Piecemeal adoption of std::ranges - adopt std::ranges for TransformOperations & FilterOperations

weinig opened the pull request:
#### 139f357261e16b23b9e26171cdf88b2e5c5b2856
<pre>
Piecemeal adoption of std::ranges - adopt std::ranges for TransformOperations &amp; FilterOperations
<a href="https://bugs.webkit.org/show_bug.cgi?id=275254">https://bugs.webkit.org/show_bug.cgi?id=275254</a>

Reviewed by Darin Adler.

Use std::ranges::any_of/std::ranges::all_of/std::ranges::equal
in TransformOperations and FilterOperations.

* Source/WebCore/platform/graphics/filters/FilterOperations.cpp:
(WebCore::FilterOperations::operator== const):
(WebCore::FilterOperations::operationsMatch const):
(WebCore::FilterOperations::hasFilterThatAffectsOpacity const):
(WebCore::FilterOperations::hasFilterThatMovesPixels const):
(WebCore::FilterOperations::hasFilterThatShouldBeRestrictedBySecurityOrigin const):
* Source/WebCore/platform/graphics/filters/FilterOperations.h:
(WebCore::FilterOperations::hasFilterOfType const):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::shouldHaveBackingStore const):
* Source/WebCore/platform/graphics/transforms/TransformOperations.cpp:
(WebCore::TransformOperations::operator== const):
(WebCore::TransformOperations::has3DOperation const):
(WebCore::TransformOperations::isRepresentableIn2D const):
(WebCore::TransformOperations::affectedByTransformOrigin const):
* Source/WebCore/platform/graphics/transforms/TransformOperations.h:
(WebCore::TransformOperations::hasTransformOfType const):

Canonical link: <a href="https://commits.webkit.org/279945@main">https://commits.webkit.org/279945@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d1685e546fe4c56c51831fb0fddd88331e99b32f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54911 "10 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34359 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7498 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58189 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5642 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57211 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41906 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5670 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44470 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3825 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57006 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32457 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47574 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25597 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29247 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4915 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3783 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51098 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5137 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59779 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30169 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5288 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51893 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31305 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47652 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51320 "Found 3 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed, /WebKitGTK/TestWebViewEditor:/webkit/WebKitWebView/select-all/editable, /WebKitGTK/TestInspectorServer:/webkit/WebKitWebInspectorServer/http-test-page-list (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12097 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32320 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31094 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/graphics/filters/FilterOperations.h:
```diff
@@ -96,6 +94,12 @@ class FilterOperations {
 };
 
+template<FilterOperation::Type Type>
+bool FilterOperations::hasFilterOfType() const
```
I typically prefer the one-line format for these, with `template` on the same line as the function signature.

Since the argument is a non-class type, can it be lowercase `type` instead of uppercase `Type`?

Commit:
Piecemeal adoption of std::ranges - adopt std::ranges for TransformOperations & FilterOperations
https://bugs.webkit.org/show_bug.cgi?id=275254

Reviewed by Darin Adler.

Use std::ranges::any_of/std::ranges::all_of/std::ranges::equal
in TransformOperations and FilterOperations.

* Source/WebCore/platform/graphics/filters/FilterOperations.cpp:
(WebCore::FilterOperations::operator== const):
(WebCore::FilterOperations::operationsMatch const):
(WebCore::FilterOperations::hasFilterThatAffectsOpacity const):
(WebCore::FilterOperations::hasFilterThatMovesPixels const):
(WebCore::FilterOperations::hasFilterThatShouldBeRestrictedBySecurityOrigin const):
* Source/WebCore/platform/graphics/filters/FilterOperations.h:
(WebCore::FilterOperations::hasFilterOfType const):
* Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayer.cpp:
(WebCore::CoordinatedGraphicsLayer::shouldHaveBackingStore const):
* Source/WebCore/platform/graphics/transforms/TransformOperations.cpp:
(WebCore::TransformOperations::operator== const):
(WebCore::TransformOperations::has3DOperation const):
(WebCore::TransformOperations::isRepresentableIn2D const):
(WebCore::TransformOperations::affectedByTransformOrigin const):
* Source/WebCore/platform/graphics/transforms/TransformOperations.h:
(WebCore::TransformOperations::hasTransformOfType const):

Canonical link: https://commits.webkit.org/279945@main
